syntax="proto3";
package users;

service Users{

    rpc GetUsers(GetUsersRequest)  returns (GetUsersResponse) {} 
    rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse) {}
    rpc CreateUser(CreateUserRequest) returns (stream CreateUserResponse) {}
    rpc DeleteUser(DeleteUserRequest) returns (DeleteUserResponse) {}
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse) {}
}

message User{
    string id=1;
    string name=2;
    string email=3;
    string password=4;
    string phone=5;
    string role_id=6;
    string created_at=7;
    string updated_at=8;
}
message GetUsersRequest {}

message GetUsersResponse{
    repeated User  users=1;
}
message GetUserByIdRequest { 		// unique id of the user to get.
    string id=1;
}

message GetUserByIdResponse { 		// user with the given id. 
 User user=1;
}

message CreateUserRequest {		// user to create.
 User user=1;
}

message CreateUserResponse {		// user with the given id.
  User user=1;
}

message UpdateUserRequest{
 User user=1;
}

message UpdateUserResponse {		
    User user=1;
}

message DeleteUserRequest{
string id=1;
}

message DeleteUserResponse {		// user with the given id.
 User user=1;
}